src/core/unit.c

```diff
@@ -4216,7 +4216,7 @@ static int user_from_unit_name(Unit *u, char **ret) {
         return 0;
 }
 
-static int unit_verify_contexts(const Unit *u, const ExecContext *ec) {
+static int unit_verify_contexts(const Unit *u, const ExecContext *ec, const KillContext *kc) {
```

Comment:
i'd probably do the reverse here: drop both ec and kc from this function prototype and instead try to get both contexts locally via unit_get_exec_context() and unit_get_kill_context().

I think that would be simpler.